src/suassuna.cpp

```diff
@@ -21,25 +21,28 @@
 
 #include "suassuna.h"
 
-Suassuna::Suassuna() {
+Suassuna::Suassuna(Constants *constants) {
+    // Setting up constants
+    _constants = constants;
+
     // Creating world
-    _world = new World();
+    _world = new World(getConstants());
 }
 
 void Suassuna::start() {
     // Creating World Map
     _worldMap = new WorldMap();
 
     // Creating and adding vision to world
-    _vision = new Vision("224.5.23.2", 10002);
+    _vision = new Vision(getConstants());
```

Comment:
When sending constants to new modules always use the **getConstants()** function, so you can **confirm** that you passed an not-null pointer and avoid crashes.